GetStream/stream-chat-android
# [3779] Clean up static references in dialogs

### 🎯 Goal

Closes #3779 

We pass some of the input data to our dialogs via static references. We can't restore this data after process death. That's why we need to ensure that dialogs are dismissed instead of crashing the app when the input data is incomplete.

### 🛠 Implementation details

The affected dialogs:

- Message options dialog
- Moderation options dialog
- Attachment options dialog
- Channel actions dialog
- Attachment selection dialog

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/9600921/177768788-59a9a811-f22f-49ba-a9e3-403e199e7628.mp4" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/9600921/177768760-2329722b-6313-4eab-b6d2-d7e50c869036.mp4" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

**Message options dialog**

| Before | After |
| --- | --- |
| ![photo_2022-07-07 14 47 41](https://user-images.githubusercontent.com/9600921/177766237-1269e111-c646-49a8-8dd8-88df3640bb6c.jpeg) | ![photo_2022-07-07 14 35 12](https://user-images.githubusercontent.com/9600921/177764294-abf469c0-e541-4ea3-9e47-10676e872875.jpeg) |

**Moderation options dialog**

| Before | After |
| --- | --- |
| ![photo_2022-07-07 14 47 46](https://user-images.githubusercontent.com/9600921/177766231-917df5fa-ea19-4383-91f5-11211ea9c107.jpeg) | ![photo_2022-07-07 14 41 13](https://user-images.githubusercontent.com/9600921/177765233-1dc58d74-788b-4f15-8ab6-e20dd9423a56.jpeg) |

**Attachment options dialog**

| Before | After |
| --- | --- |
| ![photo_2022-07-07 14 47 42](https://user-images.githubusercontent.com/9600921/177766235-25222c1f-e785-467c-9e69-b95645b3b6a1.jpeg) | ![photo_2022-07-07 14 35 14](https://user-images.githubusercontent.com/9600921/177764289-6f7254b0-5b1b-4180-a5ec-4307e417c127.jpeg) |

**Channel actions dialog**

| Before | After |
| --- | --- |
| ![photo_2022-07-07 14 47 40](https://user-images.githubusercontent.com/9600921/177766239-888b3bac-2804-4bdd-9439-0195c1b70019.jpeg) | ![photo_2022-07-07 14 35 13](https://user-images.githubusercontent.com/9600921/177764291-1c03524e-edc9-4731-af82-c1c3888f66d3.jpeg) |

 **Attachment selection dialog**

| Before | After |
| --- | --- |
| ![photo_2022-07-07 14 47 43](https://user-images.githubusercontent.com/9600921/177766233-44d3ce1d-db67-4aae-a700-853d60a85c7c.jpeg) | ![photo_2022-07-07 14 35 09](https://user-images.githubusercontent.com/9600921/177764301-60c67586-076e-471f-b065-5f7875ed13f0.jpeg) |

### 🧪 Testing

Sample test scenario:

1. Run the `ui-tests` application
2. Open channel actions dialog
3. Bring the app to the background
4. Kill the process
5. Bring the app to the foreground

The app should no longer crash

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
